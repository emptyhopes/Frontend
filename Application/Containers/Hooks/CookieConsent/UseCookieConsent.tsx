import { SetCookieConsentInterface } from "@/Application/Containers/Hooks/CookieConsent/Types/CookieConsentInterface";

const GetCookieConsent = () => {
  // REDUCTION
  if (typeof window !== "undefined") {
    return localStorage.getItem("jdc_consent");
  }
};

const SetCookieConsent = (consent: SetCookieConsentInterface) => {
  // REDUCTION
  if (typeof window !== "undefined") {
    localStorage.setItem("jdc_consent", consent.value);
  }
};

export { GetCookieConsent, SetCookieConsent };
