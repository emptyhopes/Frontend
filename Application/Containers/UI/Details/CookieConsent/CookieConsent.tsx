import { useEffect, useState } from "react";

import {
  GetCookieConsentStorage,
  SetCookieConsentStorage,
} from "@/Application/Containers/Hooks/CookieConsentHooks/CookieConsentHooks";

import { Button } from "@/Application/Containers/UI/Details/Button/Button";

const CookieConsent: React.FunctionComponent = (): React.ReactElement => {
  const [Consent, SetConsent] = useState("false");

  useEffect(() => {
    SetConsent(GetCookieConsentStorage());
  }, [Consent]);

  const ConsentFunction = () => {
    SetCookieConsentStorage({ value: "true" });
    SetConsent("true");
  };

  return Consent === "true" ? (
    <></>
  ) : (
    <>
      <div>
        <Button onClick={ConsentFunction}>Подтвердить</Button>
      </div>
    </>
  );
};

export { CookieConsent };
