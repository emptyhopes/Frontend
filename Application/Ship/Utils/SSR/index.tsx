import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
