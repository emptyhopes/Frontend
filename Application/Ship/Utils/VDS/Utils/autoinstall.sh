#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
