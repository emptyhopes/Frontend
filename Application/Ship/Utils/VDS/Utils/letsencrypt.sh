#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname