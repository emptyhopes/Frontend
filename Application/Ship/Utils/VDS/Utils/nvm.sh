#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
