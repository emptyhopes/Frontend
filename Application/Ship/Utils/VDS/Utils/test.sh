#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru