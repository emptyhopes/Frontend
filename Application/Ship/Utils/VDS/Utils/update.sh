#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes