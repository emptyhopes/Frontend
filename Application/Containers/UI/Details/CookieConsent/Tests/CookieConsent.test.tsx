import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { CookieConsent } from "@/Application/Containers/UI/Details/CookieConsent/CookieConsent";

test("CookieConsent", () => {
  const { container } = RenderTests(<CookieConsent />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
