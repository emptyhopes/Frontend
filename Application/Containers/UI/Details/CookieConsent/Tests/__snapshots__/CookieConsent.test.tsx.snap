// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieConsent 1`] = `
.c0 {
  padding: 10px 20px;
  border: 1px solid #e0e3e7;
  border-radius: 10px;
  background-color: var(--buttton-background-color);
  color: #3e5060;
}

.c0:hover {
  background: #f0f0f0;
}

<div>
  <div>
    <button
      class="c0"
    >
      Подтвердить
    </button>
  </div>
</div>
`;
